fhir-persistence-schema/docs/DB2MultiTenancy.md

```diff
@@ -0,0 +1,140 @@
+| Status | Purpose |
+|----------|---------------------------------------------|
+| PROVISIONING | Indicates the tenant schema is in the process of adding a set of tenant tables/partitions |
+| ALLOCATED | Indicates the process of creating the tenant schema is finished, and the schema is ready for use. Only allocated tenants are available for use in the `fhir-persistence-jdbc` code |
+| FROZEN | Indicates the previous tenant data is not to be used, and the tenant data is in the process of being removed |
+| DROPPED | Indicates the previous tenant data is now removed |
+| FREE | Indicates a set of tables is provisioned, and available to be allocated to a specific tenant |
+
+**Table: TENANT_KEYS**
+
+The Tenant Keys table stores a hashed version of the tenant specific key. Upon creating a session and calling the SET_TENANT procedure, the SET_TENANT queries the incoming key against the TENANT_KEYS database to map to a specific tenant MT_ID. 
+
+| Column Name | Data Type | Length | Null | Purpose |
+|----------|---------------|---------------|---------------|---------------------------------------------|
+| TENANT_KEY_ID | Integer | 4 | No | The ID is from the sequence, and is used to identify the unique entry in the row |
+| MT_ID | Integer | 4 | No | Matches to the MT_ID in the TENANTS table |
+| TENANT_SALT | VARCHAR | 44 | No | A 32 byte random value used as a salt when computing the SHA-256 tenant hash |
+| TENANT_HASH | VARBINARY | 32 | No | Each tenant's data is encrypted. |
+
+The tenant_key is a 32 byte random value created when the tenant is provisioned. This is a secret, and must be passed by the client to the database in order to obtain access to a tenant. A tenant may have multiple tenant keys to support key rotation. An example of the tenant key is `GQUHRi0ETFw0Xe+p1VtVDki4JFZs885bGM35hXRyq/E=`
+
+Multiple keys per tenant are supported, and key rotation includes adding, and then subsequently removing prior keys. 
+
+**Table: VERSION_HISTORY**
+
+The VERSION_HISTORY table enables Tenant version isolation, so upgrades and patches are separately applied to each tenant's schema. 
+
+| Column Name | Data Type | Length | Null | Purpose |
+|----------|---------------|---------------|---------------|---------------------------------------------|
+| SCHEMA_NAME | VARCHAR | 64 | No | The name of the schema where the object resides|
+| OBJECT_TYPE | VARCHAR | 16 | No | The type of the object - Procedure, Sequence, Table, TableSpace, Variable |
+| OBJECT_NAME | VARCHAR | 64 | No | The name of the object in the schema |
+| VERSION | INTEGER | 4 | No | The version starting at one and incrementing with every version change |
+| APPLIED | TIMESTAMP | 10 | No | The time of the latest object update |
+
+### Schema: Tenant-Specific
+
+The Tenant specific data is partitioned and isolated in a customer specific tablespace.  The tenant data is not mixed with other tenant data physically using the tablespace isolation. 
+
```

Comment:
does this mean we can not query the total number of Patients across all the tenants in one sql, and we have to set tenant id and query them one by one and the do the sum...

Reply:
Yes - in a multi-tenant schema.

Reply:
I've always had this question too...
Is there some way to grant cross-tenant capabilities to an admin role?
Seems like that could be useful, although I guess its designed to prevent that (for security reasons)...

Reply:
I've elaborated, as this is working as designed and intentional.  I added a paragraph explaining HOW one would accomplish it, and how the restrictions are enforced.